addons/stock/models/stock_move.py

```diff
@@ -1441,6 +1442,13 @@ def _action_done(self, cancel_backorder=False):
         for company_id, move_dests in move_dests_per_company.items():
             move_dests.sudo().with_company(company_id)._action_assign()
 
+        # if incoming moves make other confirmed/partially_available moves available, assign them
+        domains = []
+        for move in moves_todo.filtered(lambda m: m.picking_id and m.picking_id.picking_type_id.code == 'incoming'):
+            domains.append([('product_id', '=', move.product_id.id), ('location_id', '=', move.location_dest_id.id)])
+        moves_to_reserve = self.search(expression.AND([[('state', 'in', ['confirmed', 'partially_available']), ('procure_method', '=', 'make_to_stock')], expression.OR(domains)]))
+        moves_to_reserve._action_assign()
```

Comment:
can you move this block into stock_picking.py action_assign?

Reply:
@sle-odoo I moved this to _action_done, action_assign is not called when validate a picking.